### ISSUE TITLE
Custom Reporters / Printers

### ISSUE THREAD

**Issue by shellscape** [d4rkr00t/whybundled#19]
(You might have noticed that I'm enthusiastic about your project :smile:)

While the default output is definitely colorful, distinct, and pleasant to look at, I'd like to request the ability to use a custom reporter / printer for controlling the styles applied to the output. I'd like to use this as a component in a project that follows the "stylish conventions" and a custom reporter would help me to keep the output for `whybundled` in line with that style.

**Comment by d4rkr00t** [comment]
Yeah, it definitely makes sense, I'll have a look at adding custom print functions asap.

**Comment by d4rkr00t** [comment]
I've been thinking about that, and even though it's not hard to add a custom print function, I want to better understand your use case. 

I don't think it's a good idea to use `commands/{default|by}.js` as an Node API for `whybundled`, because these functions handle lots of cli specific things e.g. progress bar, reading files from fs, killing the process in case something goes wrong, etc... I can extract non-cli specific stuff from commands and provide them as an API, which seems totally reasonable. 

But if you don't need the logic inside commands, if default behaviour is enough, you can use `analyze` and build your own print:

```js
const { analyze } = require('whybundled');
const report = analyze(stats);
customPrint(report);
```

**Comment by d4rkr00t** [comment]
Oh, ok :) Cool! It shouldn't be much work to add that. I'll try and put up a PR in couple of days.

**Comment by d4rkr00t** [comment]
Published as [v1.4.0](https://github.com/d4rkr00t/whybundled/releases/tag/v1.4.0)
